### PR TITLE
Fixes gradle error: 'Could not find lint-gradle-api.jar' (#23095)

### PR DESCRIPTION
**Pull request by mrmitew**
Fixes gradle error:

Could not resolve all files for configuration 'classpath'.
Could not find lint-gradle-api.jar (com.android.tools.lint:lint-gradle-api:26.1.2).
Searched in the following locations:
https://jcenter.bintray.com/com/android/tools/lint/lint-gradle-api/26.1.2/lint-gradle-api-26.1.2.jar

### CODE REVIEW

**Comment by mrmitew** [comment]
I signed it!

**Review by ngenovictor — APPROVED** [review]
After changing the line. My app was able to run.

**Comment by Mahi-K** [comment]
Hi, 
I've added `google()` inside the `buildScript` and also updated the gradle-wrapper properties distributionUrl to `distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip`. 

change the buildScript from `$flutterRoot/packages/flutter_tools/gradle/flutter.gradle` file to 

```
buildscript {
    repositories {
        google()
        jcenter()
        maven {
            url 'https://dl.google.com/dl/android/maven2'
        }
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.2.1'
    }
}
```

and the issue disappeared for me . try these if any one is facing the same issue.

**Comment by ScholliYT** [comment]
any ideas on how I can get travis to use this fix too ? ^^

**Comment by Ronshaoshu** [comment]
it will be noticed that the` google()` should be joined in : 
 `**flutter/packages/flutter_tools/gradle/flutter.gradle**`
it spends me almost 1 day to solve......

**Comment by nathansamson** [comment]
Pinging @Hixie @tvolkert @gspencergoog 

This is an isuse that is breaking lots of people CI + local build systems
Any chance this fix can be merged sooner than rather + backported to the latest beta version?

**Comment by gspencergoog** [comment]
Looks like we forgot got re-land https://github.com/flutter/flutter/pull/13565 after the bots had accepted the new license. @dnfield, do you feel Gradle-aware enough to land that?

File: packages/flutter_tools/gradle/flutter.gradle [hunk 1]
```diff
@@ -18,6 +18,7 @@ import org.gradle.api.tasks.bundling.Jar
 
 buildscript {
     repositories {
+        google()
         jcenter()
         maven {
```
Comment (dnfield) [inline packages/flutter_tools/gradle/flutter.gradle]: The entire `maven {...}` should be removed.  It looks like our problem is that jcenter is being used first, and we want to use google.

I think it's fine to use `google()` at this point, as we're not using gradle versions old enough to not know about it, but having the redudancy in here doesn't make sense to me.

**Comment by dnfield** [comment]
Just have one comment on this one and then I'm comfortable with landing it.  

Do we need to update the device lab templates in the same PR?

**Comment by gspencergoog** [comment]
I think it would be good to keep them in sync, and obviously Mikkel did too, so I'd say yes, but if you want to split them out, that might make it easier to revert if needed.

**Comment by charafau** [comment]
duplicate of https://github.com/flutter/flutter/pull/23467

**Comment by ScholliYT** [comment]
For everyone having trouble with travis: I used this "hack" to get travis working again.


```
before_script:
  - git clone https://github.com/flutter/flutter.git -b beta --depth 1
  - export PATH=./flutter/bin:$PATH  
  - cd ./flutte
  - git fetch origin pull/23397/head:pull_23397
  - git checkout pull_23397
  - cd ..
```

It's just checking out this pull request and using that code to run.

**Comment by ffeu** [comment]
Apparently, PRs to flutter/plugins are not passing exactly because of this problem.
The last 3 PRs (867, 868, flutter/plugins#869) are stuck with this error.

**Comment by Joaotiago87** [comment]
Total newb here. I'm having this issue. How do I solve it? 
Can I add `Google()` to my project? Do I have to wait for an update?
Any help appreciated.

**Comment by Joaotiago87** [comment]
> For everyone having trouble with travis: I used this "hack" to get travis working again.
> 
> ```
> before_script:
>   - git clone https://github.com/flutter/flutter.git -b beta --depth 1
>   - export PATH=./flutter/bin:$PATH  
>   - cd ./flutter
>   - git fetch origin pull/23397/head:pull_23397
>   - git checkout pull_23397
>   - cd ..
> ```
> It's just checking out this pull request and using that code to run.

where do I execute that?

**Comment by dnfield** [comment]
LGTM, made the change myself since this is getting a lot of requests.

**Comment by dnfield** [comment]
Will land on green.

**Comment by sultanmyrza** [comment]
> Hi,
> I've added `google()` inside the `buildScript` and also updated the gradle-wrapper properties distributionUrl to `distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip`.
> 
> change the buildScript from `$flutterRoot/packages/flutter_tools/gradle/flutter.gradle` file to
> 
> ```
> buildscript {
>     repositories {
>         google()
>         jcenter()
>         maven {
>             url 'https://dl.google.com/dl/android/maven2'
>         }
>     }
>     dependencies {
>         classpath 'com.android.tools.build:gradle:3.2.1'
>     }
> }
> ```
> and the issue disappeared for me . try these if any one is facing the same issue.

The solution above worked for me I have 
[✓] Flutter (Channel beta, v0.9.4, on Mac OS X 10.13.4 17E202, locale en-KR)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
[✓] Android Studio (version 3.1)
[✓] VS Code (version 1.28.2)
[✓] Connected devices (1 available)

This error will go after this https://github.com/flutter/flutter/pull/23397 merge
but for now, just use instruction above.
Thanks @Mahi-K

**Comment by 000lsp123** [comment]
> 可以了,谢谢.

how resolve?

**Comment by fe9lsh3ben** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here (e.g. `I signed it!`) and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google.com/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).

--------

I followed of those steps and still doesn't work with me.

Any other solution ??

**Comment by cmonney** [comment]
This solution is not working for me too

**Comment by bmatheus91** [comment]
> Hi,
> I've added `google()` inside the `buildScript` and also updated the gradle-wrapper properties distributionUrl to `distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip`.
> 
> change the buildScript from `$flutterRoot/packages/flutter_tools/gradle/flutter.gradle` file to
> 
> ```
> buildscript {
>     repositories {
>         google()
>         jcenter()
>         maven {
>             url 'https://dl.google.com/dl/android/maven2'
>         }
>     }
>     dependencies {
>         classpath 'com.android.tools.build:gradle:3.2.1'
>     }
> }
> ```
> and the issue disappeared for me . try these if any one is facing the same issue.

Thank you, this solved! Besides those changes i did this: 
2018-09-21_20-24-19-94c074f02fcb97ecd84bb33aa57cb86c

**Comment by fe9lsh3ben** [comment]
I just solve the issue thinks to Allah then to then to the 24 contributors who solved this issue 

At first I don't know if the first step is necessary to solve this issue or not, but I did it.

first step :-
- I  visited this link https://cla.developers.google.com/ to sign as I understood from googlebot ,with adding my github user name.
- thin I followed the steps in this link ( https://help.github.com/articles/setting-your-commit-email-address-in-git/ )to link my github account with the git command, to be honest I didn't understand why am doing this think but I just tried follow what googlebot said to fix this problem.
--------------------------

second step :- 
-Go the fluuter.gradle from the flutter that you have already downloaded it.
In my cause I downloaded it in C:\src
C:\src\flutter\packages\flutter_tools\gradle\fluuter.gradle
- Copy the codes from this link  and replace it with exist code in this file (fluuter.gradle).

https://github.com/mrmitew/flutter/blob/6369f5481c08714b8a724a92dea2328d23d7bba3/packages/flutter_tools/gradle/flutter.gradle

one day and half to solve the problem :(

**Comment by zerox1212** [comment]
Thanks for the merge. Lost at least 4 hours trying to figure this when trying to test out flutter.

**Comment by dustin-graham** [comment]
I'd like to say thanks to those who have been working to resolve this. However, I can't see in this thread  any identification of the root cause. My CI builder was able to build my project last week on the same version of flutter and without any local code changes. Did the external dependency move? If so, it would seem wise to see if we could get it put back where it was until this lands in Beta. Does anyone know what happened to cause this?

**Comment by gamebox** [comment]
I have the following version:

```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v0.9.4, on Linux, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✓] Android Studio (version 3.2)
[✓] VS Code (version 1.28.2)
[✓] Connected devices (1 available)

```
Looking in build.gradle shows the change from the referenced PR applied, but I am still getting the same error.  Seems like it doesn't even check the google repository.

**Comment by diegoveloper** [comment]
> I have the following version:
> 
> ```
> $ flutter doctor
> Doctor summary (to see all details, run flutter doctor -v):
> [✓] Flutter (Channel beta, v0.9.4, on Linux, locale en_US.UTF-8)
> [✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
> [✓] Android Studio (version 3.2)
> [✓] VS Code (version 1.28.2)
> [✓] Connected devices (1 available)
> ```
> Looking in build.gradle shows the change from the referenced PR applied, but I am still getting the same error. Seems like it doesn't even check the google repository.

I think it's not the build.gradle file, the name of the file is flutter.gradle inside this directory : 

      $flutterRoot/packages/flutter_tools/gradle/

**Comment by gamebox** [comment]
I believe that's the root of the flutter project, and that is what creates the app build.gradle for an application when using `flutter create` @diegoveloper

**Comment by diegoveloper** [comment]
then I have the same issue :/

**Comment by gamebox** [comment]
@diegoveloper Are you also on Linux?  Could you share the output of `flutter doctor`?

**Comment by diegoveloper** [comment]
Actually I don't have any issues in my local machine (I'm using Mac and Flutter v0.10.1) 

			[✓] Flutter (Channel dev, v0.10.1, on Mac OS X 10.14 18A391, locale en-PE)
			[✓] Android toolchain - develop for Android devices (Android SDK 28.0.0)
			[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
			[✓] Android Studio (version 3.1)
			    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
			    ✗ Dart plugin not installed; this adds Dart specific functionality.
			[✓] VS Code (version 1.28.2)

The problem I have is on CI environment, It's on Linux

			[✓] Flutter (Channel dev, v0.10.1, on Linux, locale en_US)
			    • Flutter version 0.10.1 at /opt/flutter
			    • Framework revision 6a3ff018b1 (8 days ago), 2018-10-18 18:38:26 -0400
			    • Engine revision 3860a43379
			    • Dart version 2.1.0-dev.7.1.flutter-b99bcfd309

   I know the PR was merge into master channel, but I'm using dev channel

**Comment by gamebox** [comment]
Interesting.  Well I've tried everything I can think of for tonight.  I'm just going to write some Dart, and hope I can figure it out soon

**Comment by AnthonyRobertson17** [comment]
> @nathansamson I'm not sure about merging this into the latest beta, but it is possible to fix this manually by modifying build.gradle. See, for example, this commit: https://github.com/flutter/flutter/pull/23438/files#diff-89970c613560f892af5f6c393a52d2c0R3
> 
> (Unless your CI is regenerating that file, which I think should only be happening if you're testing `flutter create` for some reason.)

I am working on the beta channel and I applied the manual fix but still get the same errors. The cause for me seems to stem from the following line `apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"` which resides in my `android/app/build.gradle`. Apply the same manual fix to `flutter.gradle` then fixes the issue, but that's just the same as me manually back-porting the fix in this PR. Being new to flutter I tried removing the line which applies `flutter.gradle` thinking it might not be needed for non demo apps, but I just got more errors. Not really sure what to try from here other than just leaving the `flutter.gradle` modified until this change makes it into the beta channel.

**Comment by Mahi-K** [comment]
@zoechi I think yes, there is actually no need to merge this, although we need to upgrade the gradle version to 3.2.1 and gradle wrapper to 4.6, update all the firebase plugins to the latest versions so that there are no performance issues (as my app was crashing most of the time because of FirebaseMessaging). Thanks.

**Comment by koral--** [comment]
> there is actually no need to merge this, although we need to upgrade the gradle version to 3.2.1 and gradle wrapper to 4.6, (...)
The file at "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle" can be changed as
(...)

There are few things confused here: 3.2.1 is not a version of Gradle but Android Gradle plugin. Secondly, `maven { url 'https://dl.google.com/dl/android/maven2'  }` has the same effect as `google()` since Gradle (wrapper) 4.1. https://developer.android.com/studio/build/dependencies#google-maven

**Comment by patientplatypus** [comment]
Can someone be more explicit on where this buildscript file is supposed to be found? "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle" does not exist in a create flutter boiler plate app and changing the buildscript in build.gradle does not fix this issue. A little frustrating that hello world no longer works -_-.

**Comment by Mahi-K** [comment]
Hi sorry about the experience, you can find this file from your flutter directory which is usually in 
`C:/Users/name/flutter/packages/flutter_tools/gradle/flutter.gradle`  and on mac 
`/Users/yourname/development/flutter/packages/flutter_tools/gradle/flutter.gradle` if you have installed it to development folder or `/Users/yourname/flutter...` you can simply use IJIDEAEditor to edit the file or any notepad or texteditor will suffice.

Hope this helps thanks.

**Comment by Mahi-K** [comment]
> > there is actually no need to merge this, although we need to upgrade the gradle version to 3.2.1 and gradle wrapper to 4.6, (...)
> > The file at "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle" can be changed as
> > (...)
> 
> There are few things confused here: 3.2.1 is not a version of Gradle but Android Gradle plugin. Secondly, `maven { url 'https://dl.google.com/dl/android/maven2' }` has the same effect as `google()` since Gradle (wrapper) 4.1. https://developer.android.com/studio/build/dependencies#google-maven

sorry mybad, i meant gradle plugin, yes maven's repository now means google(). Thanks

**Comment by patientplatypus** [comment]
Thanks Mahi-k - I can confirm your fix works!

**Comment by Mahi-K** [comment]
Thank you @patientplatypus @gamebox glad that it worked for you. Happy coding 👍

**Review by sirdlx — COMMENTED** [review]
Worked for me!

**Comment by mrmitew** [comment]
> Does [#23404 (comment)](https://github.com/flutter/flutter/issues/23404#issuecomment-433669581) mean this PR should be undone?

JCenter proxies the central repository and provides additional dependencies. However, proxying seems to be broken quite regularly (there were also long threads in Twitter about it not long time ago). Its better to use google's repository directly for any artifacts that are available there and fallback to jcenter for those that do not exist. Many times in order to fix a gradle build, adding google's repo as a first option, is the way to go.

**Comment by bucc-sanyam** [comment]
I am really confused by all the comments above. Can someone please guide me how to resolve this problem. I am trying to resolve this error for past 4 hrs and am unable to do so.

**Comment by bucc-sanyam** [comment]
> Hi sorry about the experience, you can find this file from your flutter directory which is usually in
> `C:/Users/name/flutter/packages/flutter_tools/gradle/flutter.gradle` and on mac
> `/Users/yourname/development/flutter/packages/flutter_tools/gradle/flutter.gradle` if you have installed it to development folder or `/Users/yourname/flutter...` you can simply use IJIDEAEditor to edit the file or any notepad or texteditor will suffice.
> 
> Hope this helps thanks.

Thank You so much man!

File: AUTHORS [hunk 2]
```diff
@@ -28,3 +28,4 @@ Christian Mürtz <teraarts@t-online.de>
 Lukasz Piliszczuk <lukasz@intheloup.io>
 Felix Schmidt <felix.free@gmx.de>
 Artur Rymarz <artur.rymarz@gmail.com>
+Stefan Mitev <mr.mitew@gmail.com>
```
Comment (krutartha) [inline AUTHORS]: Krutartha Nagesh <krutartha2002@gmail.com>

**Comment by elrasool** [comment]
Unfortunately, nothing of all solutions above has solved the problem for me, neither (4.10.2), gradle 3.2.1, nor google() above jcenter().

**Comment by rogercotrina** [comment]
@elrasool It should be fixed in the flutter beta channel. Ensure you are in there by typing `flutter channel`, and if in beta, just run `flutter upgrade` to get latest changes. See https://github.com/flutter/flutter/pull/23397 for further info.

**Comment by Andrewx987** [comment]
Guys this is a workaround actually working for fixing flutter issues with gradle :  


1) First, important!!, reimport the folder "app" while creating flutter proyect!!

2) Modify exactly the gradle with the version 3.3.2  and the  .yaml  file not exceed version more than >= 1.0.0

3) Go to android/build.gradle and copy exactly this:


``
buildscript {
    repositories {
        google()
        jcenter()
        mavenCentral()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:3.3.2' (This is crucial for work)
        classpath 'com.google.gms:google-services:4.2.0'
    }

}
    allprojects {
        repositories {
            google()
            jcenter()
            mavenCentral()
        }
    }

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

task clean(type: Delete) {
    delete rootProject.buildDir
}

subprojects {
    project.configurations.all {
        resolutionStrategy.eachDependency { details ->
            if (details.requested.group == 'com.android.support'
                    && !details.requested.name.contains('multidex') ) {
                details.useVersion "27.1.0"
            }
        }
    }
}

//resolutionStrategy.eachDependency { details ->
//    if (details.requested.group == 'com.android.support'
//            && !details.requested.name.contains('multidex') ) {
//        details.useVersion "26.1.0"
//    }
//} ``


4) Ready!!

**Pull request merged**